test/test_tornado.py

```diff
@@ -43,21 +44,33 @@ def __fetch(self, url: str):
         self.__io_loop.run_sync(lambda: future)
         return future.result()
 
-    def _get_response(self, url: str, benchmark: Callable) -> BaseServer.Response:
+    def _get_response(self, url: str, benchmark: Callable) -> Response:
         r = benchmark(lambda: self.__fetch(url))
-        return BaseServer.Response(
+        return Response(
             status=r.code, headers=list(r.headers.get_all()), content=r.body
         )
 
     def assert_404(self, url: str):
-        with pytest.raises(HTTPClientError) as e:
+        with pytest.raises(HTTPClientError) as exc:
             self._get_response(url, lambda f: f())
-        assert e.value.code == 404
+        r = exc.value.response
+        assert r is not None
```

Comment:
A bit unfortunate that I have to get the response from the thrown exception, preventing me to use the `assert_404` defined in `BaseServer`...